app/main.py

```diff
@@ -1 +1,34 @@
-# write your code here
+class Animal:
+
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, animal: Herbivore) -> None:
+        if not isinstance(animal, Carnivore) and not animal.hidden:
+            animal.health -= 50
+        if animal.health <= 0:
+            Animal.alive.remove(animal)
+            del animal
```

Comment:
I'm sorry, I don't understand why you delete it again, maybe it ok, sorry

Reply:
`Animal.alive.remove(animal)` here I delete it from list alive and `del animal` I delete object because it's dead

Reply:
Ok, good job!!!